prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,105 @@
+  * Standard Operating Procedure (SOP) style documentation for resolving alerts is required. (SOP Template Placeholder - https://issues.redhat.com/browse/SIGSRE-55)
+
+## Critical Alerts
+
+Critical alerts are for alerting current and impending disaster situations.  These alerts page an SRE.  The situation should warrant waking someone in the middle of the
+night.
+
+Timeline: ~5 minutes.
+
+Reserve critical level alerts only for reporting conditions that may lead to service unavailability.
+Failures of individual components should not trigger critical level alerts in general, unless they would result in that condition.
+Configure critical level alerts so they fire before the situation becomes irrecoverable.
+
+Example critical alert:
+
+```yaml
+- alert: BackendServiceStuck
+  expr: backend_state != 1
+  for: 10m
+  labels:
+    severity: critical
+  annotations:
+    summary: 'The backend service is stuck in a non-ready state'
+    description: 'The backend service {{ $labels.name }} in the {{ $labels.namespace }} namespace, managed by operator {{ $labels.pod }} has been in a non-ready state for 10 minutes'
+    sop_url: 'https://example.com/backend_service_stuck.asciidoc'
+```
+
+This alert fires if a backend service has *not* been ready for the last 10 minutes.
+This is a clear example of a critical issue that represents a threat to the operability of the service, and likely warrants paging someone.
+The alert has a clear summary and description annotations, and it links to a SOP with information on investigating and resolving the issue.
+
+The group of critical alerts should be small, very well-defined, highly documented, polished and with a high bar set for entry.
+
+## Warning Alerts
+
+The vast majority of alerts should use this severity.
+Issues at the warning level should be addressed in a timely manner, but don't pose an immediate threat to the operation of the service as a whole.
+
+Timeline: ~60 minutes
+
+If your alert does not meet the criteria in "Critical Alerts" above, it belongs to the warning level or lower.
+
+Use warning level alerts for reporting conditions that may lead to inability to deliver individual features of the service, but not the service as a whole.
+Most alerts are likely to be warnings.
+Configure warning level alerts so that they do not fire until components have sufficient time to try to recover from the interruption automatically.
+You should have a policy for triaging warning alerts and responding in a timely manner e.g. issues created automatically on a backlog that is being actively worked from.
+
+Example warning alert:
+
+```yaml
+- alert: ZookeeperPersistentVolumeFillingUp
+  expr: (kubelet_volume_stats_available_bytes{persistentvolumeclaim=~"data-(.+)-zookeeper-[0-9]+"} / kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~"data-(.+)-zookeeper-[0-9]+"} < 0.15) and predict_linear(kubelet_volume_stats_available_bytes{persistentvolumeclaim=~"data-(.+)-zookeeper-[0-9]+"}[6h], 4 * 24 * 3600) < 0
```

Comment:
I might pick something with a less complex PromQL expression, this one is enough to make someone go "wait what?" 😉 (The focus should be on the structure of the YAML, not so much on the expression inside of it.)

Reply:
Agreed. 
I'll change this to a simpler expression that shouldn't shift focus away from what's important in this doc i.e. consistency